apps/frontend/src/renderer/components/Insights.tsx

```
@@ -415,15 +543,210 @@ function MessageBubble({
+                      <div className="flex items-center justify-between bg-muted/30 border-t border-x border-border rounded-t-lg px-4 py-2">
+                        {language && (
+                          <span className="text-xs font-medium text-muted-foreground">
+                            {language}
+                          </span>
+                        )}
+                        <button
+                          onClick={handleCopy}
+                          className="ml-auto flex items-center gap-1.5 px-2 py-1 text-xs text-muted-foreground hover:text-foreground transition-colors rounded"
+                          title={t('code.copy')}
+                        >
+                          {copied ? (
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)